### PR TITLE
Add support for goma compiler when building LLVM

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: src/build.py [hunk 1]
```diff
@@ -691,17 +691,28 @@ def LLVM():
...
                       c for c in ['C', 'CXX']])
 
+  if compiler_launcher:
+    command.extend(['-DCMAKE_%s_COMPILER_LAUNCHER=%s' %
+                    (c, compiler_launcher) for c in ['C', 'CXX']])
+
   proc.check_call(command, cwd=LLVM_OUT_DIR)
-  proc.check_call(['ninja', '-v'], cwd=LLVM_OUT_DIR)
+  proc.check_call(['ninja', '-v'] + jobs, cwd=LLVM_OUT_DIR)
+  sys.exit(1)
```
Comment (dschuff) [inline src/build.py]: I assume this was for local testing? :)

**Comment by kripken** [comment]
Just curious, what's the goma compiler?

File: src/build.py (hunk 1 above)
Comment (sbc100) [inline src/build.py]: Doh! Thanks

**Comment by dschuff** [comment]
It's a distributed compilation infrastructure used by Chromium's buildbots. Basically it's just a big distcc that runs on all 3 OSes. Unfortunately I don't think it's available outside of Google, so we still have the ccache fallback (and of course the non-cached fallback).

**Comment by sbc100** [comment]
goma is the distributed compiler used by chromium that performs compiles on GCE and has aggressive caching.  Right now its only available to google employees, but its also available on the builtbots which means it can be utilized there to speed up the build.  I'll update the CL description.

File: src/build.py [hunk 2]
```diff
@@ -691,17 +691,27 @@ def LLVM():
...
              '-DLLVM_TARGETS_TO_BUILD=X86']
-  ccache = Which('ccache')
-  if ccache:
-      command.extend(['-DCMAKE_%s_COMPILER_LAUNCHER=%s' %
-                      (c, ccache) for c in ['C', 'CXX']])
+
+  compiler_launcher = None
+  jobs = []
+  if 'GOMA_DIR' in os.environ:
+    compiler_launcher = os.path.join(os.environ['GOMA_DIR'], 'gomacc')
```
Comment (dschuff) [inline src/build.py]: Do we need to run `goma_ctl.py ensure_start` or some such, or does the bot infra do that?

**Comment by kripken** [comment]
Interesting, thanks for the info.

File: src/build.py (hunk 2 above)
Comment (dschuff) [inline src/build.py]: Anyway I'm about to be gone for the day so LGTM if it's unnecessary or if it is necessary and you add it :)

Comment (sbc100) [inline src/build.py]: No, I think the bot setup takes care of all that.

Comment (dschuff) [inline src/build.py]: Hm, maybe not? https://wasm-stat.us/builders/linux/builds/11019/steps/LLVM/logs/stdio

Comment (sbc100) [inline src/build.py]: Arg.  Will fix.  Want me to revert this for now?

**Pull request merged**